tasks/update-cmr-access-constraints/index.js

```diff
@@ -15,6 +16,34 @@ const {
   uploadUMMGJSONCMRFile,
 } = require('@cumulus/cmrjs/cmr-utils');
 
+function createUpdatedEcho10XMLMetadataGranuleCopy(metadataGranule, accessConstraintsObject) {
```

Comment:
I am kind of confused as to what's going on here. I thin a function comment would help.

Reply:
Echo10XML follows a schema that includes a specific ordering for keys, so we have to follow that ordering as well, lest we fail CMR validation. I'll try to explain by adding a function comment.

Reply:
I am really worried about this function and whatever configurations of Echo10 we could get here. I'm kind of worried we're not testing it enough, but I'm not really sure what to do about all of it. I ran into this same issue when I was on CMR.

I looked at what we did for OnlineAccessURLs and I think we're making the assumption that they have them already, which we really can't do here, so I guess this is the way it has to be.

Reply:
Based on the [Echo10XML schema](https://git.earthdata.nasa.gov/projects/EMFD/repos/echo-schemas/browse/schemas/10.0/Granule.xsd), there are no additional fields that I haven't covered, as there are only a few before the `RestrictionX` fields. Since it uses the `<sequence>` tag I'm pretty comfortable with the output from this function not failing validation, barring and schema updates. But I doubt Echo10XML will see schema changes given its age.

What gives me a lot more discomfort is using the XML library in this way. That seems much more likely to receive a breaking change or another by comparison. But there is very little we can do short of writing a complete Echo10XML generator.

OnlineAccessUrls has the benefit of being more distinctly separate in terms of schema elements, but you are correct in that our current code assumes they already exist. I'm not sure it would work if they did not.

Reply:
Open to a veto and feedback from team (cc @markdboyd @Jkovarik @yjpa7145).

Reply:
This should probably have a unit test set

Reply:
Also, again, not your PR (and not that you don't know this)  but the lack of typing/doc strings in here makes this fun to weed through.   We should probably put this high on the TS candidate list.

Reply:
Aside possible code nits, I think 'writing a complete Echo10XML generator' is outside of scope - if we have this documented via unit tests... it works.

Reply:
The question is what to unit test against? I tried for a while but couldn't get testing against the XSD to behave, and it feels like submitting it to the CMR validate endpoint brings in tons of dependencies and is also not very unit-testy.

Reply:
This is my second PR for this ticket, I did create this task, so note taken about docstrings. I'm not sure that we are completely settled on how much we put docstrings on stuff that is not exported. I tried to name effectively, at least.

I tried to implement the task in TS, but that would have required converting cmrjs to TS, which... gave me nightmares.

Reply:
> The question is what to unit test against?

Hmm, let's chat tomorrow -  I think we can test using our own internal assumptions to guard against things like breaking changes from the xml library dependency, etc, but I feel like I'm misunderstanding something based on your comment.